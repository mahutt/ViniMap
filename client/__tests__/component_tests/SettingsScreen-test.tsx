import React from 'react';
import { render } from '@testing-library/react-native';
import SettingsScreen from '@/app/(tabs)/settings/index';
import { Stack } from 'expo-router';

// Mock the expo-router Stack component
jest.mock('expo-router', () => ({
  Stack: {
    Screen: jest.fn().mockReturnValue(null),
  },
}));

// Mock the child components
jest.mock('@/components/SettingsItem', () => {
  return jest.fn().mockImplementation(({ title, route }) => {
    return null;
  });
});

jest.mock('@/components/RecordingButton', () => {
  return jest.fn().mockImplementation(() => {
    return null;
  });
});

describe('SettingsScreen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with all components', () => {
    const { toJSON } = render(<SettingsScreen />);

    expect(toJSON()).toMatchSnapshot();

    expect(Stack.Screen).toHaveBeenCalledWith(
      expect.objectContaining({
        options: {
          title: 'Settings',
          headerShadowVisible: false,
          headerStyle: {
            backgroundColor: '#FFFFFF',
          },
        },
      }),
      {}
    );

    const SettingsItem = require('@/components/SettingsItem');
    const RecordingButton = require('@/components/RecordingButton');

    expect(SettingsItem).toHaveBeenCalledTimes(3);
    expect(SettingsItem).toHaveBeenCalledWith(
      { title: 'Language', route: '/settings/language' },
      {}
    );
    expect(SettingsItem).toHaveBeenCalledWith(
      { title: 'Text Size', route: '/settings/textsize' },
      {}
    );
    expect(SettingsItem).toHaveBeenCalledWith(
      { title: 'Color Blindness', route: '/settings/colorblind' },
      {}
    );

    expect(RecordingButton).toHaveBeenCalledTimes(1);
  });

  it('has the correct styles defined', () => {
    const { toJSON } = render(<SettingsScreen />);
    expect(toJSON()).toBeTruthy();
  });
});
