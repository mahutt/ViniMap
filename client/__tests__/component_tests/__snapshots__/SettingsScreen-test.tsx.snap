// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsScreen renders correctly with all components 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "gap": 16,
        "justifyContent": "center",
        "paddingHorizontal": 16,
      }
    }
  />
</View>
`;
